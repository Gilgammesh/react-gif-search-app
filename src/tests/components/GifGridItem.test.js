import React from "react";
import { shallow } from "enzyme";
import "@testing-library/jest-dom";
import GifGridItem from "../../components/GifGridItem";

describe("Pruebas unitarias en el componente GifGridItem.js", () => {
  const title = "un titulo";
  const url = "https://media1.giphy.com/test.gif";
  const classTest = "animate__fadeIn";
  /************************************************************************************/
  test("se debe mostrar el componente <GifGridItem /> correctamente", () => {
    const wrapper = shallow(<GifGridItem url={url} title={title} />);
    expect(wrapper).toMatchSnapshot();
  });
  /************************************************************************************/
  test("se debe tener un parrafo con titulo", () => {
    const wrapper = shallow(<GifGridItem url={url} title={title} />);
    const parraf = wrapper.find("p");
    expect(parraf.text().trim()).toBe(title);
  });
  /************************************************************************************/
  test("se debe tener la imagen una url y alt", () => {
    const wrapper = shallow(<GifGridItem url={url} title={title} />);
    const img = wrapper.find("img");
    expect(img.props().alt).toBe(title);
    expect(img.props().src).toBe(url);
  });
  /************************************************************************************/
  test("se debe tener en el div principal la clase animate__fadeIn", () => {
    const wrapper = shallow(<GifGridItem url={url} title={title} />);
    const div = wrapper.find("div");
    const className = div.props().className;
    expect(className.includes(classTest)).toBe(true);
  });
});
