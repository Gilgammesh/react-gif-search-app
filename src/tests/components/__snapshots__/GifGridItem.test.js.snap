// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas unitarias en el componente GifGridItem.js se debe mostrar el componente <GifGridItem /> correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="un titulo"
    src="https://media1.giphy.com/test.gif"
  />
  <p>
    un titulo
  </p>
</div>
`;
